common/src/legacy_memory_region.rs

```diff
@@ -306,3 +343,276 @@ where
         None
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use bootloader_api::info::MemoryRegionKind;
+
+    #[derive(Copy, Clone, Debug)]
+    struct TestMemoryRegion {
+        start: PhysAddr,
+        len: u64,
+        kind: MemoryRegionKind,
+    }
+
+    impl LegacyMemoryRegion for TestMemoryRegion {
+        fn start(&self) -> PhysAddr {
+            self.start
+        }
+
+        fn len(&self) -> u64 {
+            assert!(self.len % 4096 == 0);
+            self.len
+        }
+
+        fn kind(&self) -> MemoryRegionKind {
+            self.kind
+        }
+
+        fn usable_after_bootloader_exit(&self) -> bool {
+            match self.kind {
+                MemoryRegionKind::Usable => true,
+                _ => false,
+            }
+        }
+    }
+
+    // we need some kind of max phys memory, 4GB seems reasonable
+    const MAX_PHYS_ADDR: u64 = 0x4000_0000;
+
+    fn create_single_test_region() -> Vec<TestMemoryRegion> {
+        vec![TestMemoryRegion {
+            start: PhysAddr::new(0),
+            len: MAX_PHYS_ADDR,
+            kind: MemoryRegionKind::Usable,
+        }]
+    }
+
+    #[test]
+    fn test_all_regions_frame_alligned() {
+        let regions = create_single_test_region();
+        let mut allocator = LegacyFrameAllocator::new(regions.into_iter());
+        // allocate at least 1 frame
+        allocator.allocate_frame();
+
+        let mut regions = [MaybeUninit::uninit(); 10];
+        let kernel_slice_start = PhysAddr::new(0x50000);
+        let kernel_slice_len = 0x0500;
+        let ramdisk_slice_start = None;
+        let ramdisk_slice_len = 0;
+
+        let kernel_regions = allocator.construct_memory_map(
+            &mut regions,
+            kernel_slice_start,
+            kernel_slice_len,
+            ramdisk_slice_start,
+            ramdisk_slice_len,
+        );
```

Comment:
This test fails, because the kernel slice len is not 4k in size. I assume that is the case with the current implementation as well. I have not yet tested this, but I don't have more time today.